crates/ruma-client-api/src/state/send_state_event.rs

```diff
@@ -197,9 +229,35 @@ pub mod v3 {
                     (a, b, "".into())
                 };
 
+            #[cfg(feature = "unstable-msc3316")]
+            let request_query: RequestQuery =
+                ruma_common::serde::urlencoded::from_str(request.uri().query().unwrap_or(""))?;
+
             let body = serde_json::from_slice(request.body().as_ref())?;
 
-            Ok(Self { room_id, event_type, state_key, body })
+            Ok(Self {
+                room_id,
+                event_type,
+                state_key,
+                body,
+                #[cfg(feature = "unstable-msc3316")]
+                timestamp: request_query.timestamp,
+            })
         }
     }
+
+    /// Data in the request's query string.
+    #[cfg(feature = "unstable-msc3316")]
+    #[derive(Debug)]
+    #[cfg_attr(feature = "client", derive(Serialize))]
+    #[cfg_attr(feature = "server", derive(Deserialize))]
```

Comment:
```suggestion
    #[cfg_attr(feature = "client", derive(serde::Serialize))]
    #[cfg_attr(feature = "server", derive(serde::Deserialize))]
```

then you don't need the extra imports.

Reply:
Right :sweat_smile: